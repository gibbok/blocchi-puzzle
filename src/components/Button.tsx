import styled from 'styled-components';
import React from 'react';

const ButtonStyled = styled.button`
  cursor: pointer;
  border: none;
  display: block;
  font-size: clamp(3rem, 5vmin, 7rem);
  min-height: 6rem;
  color: white;
  background: linear-gradient(
    0deg,
    rgba(50, 140, 0, 1) 0%,
    rgba(62, 171, 3, 1) 25%,
    rgba(69, 188, 4, 1) 85%,
    rgba(63, 173, 2, 1) 100%
  );
  /* border-radius: 0.6rem; */
  border-radius: 1vmin;
  box-shadow: 0 0.4rem 0.7rem 0 #165801, inset 0 0 0.4rem 0 #1a6f00;
  &:focus {
    outline: none;
  }
`;

const Content = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 1.4rem;
`;

type Props = Readonly<{
  children?: React.ReactNode;
  onClick: () => void;
}>;

export function Button({ children, onClick }: Props): JSX.Element {
  return (
    <ButtonStyled onClick={onClick}>
      <Content>{children}</Content>
    </ButtonStyled>
  );
}
