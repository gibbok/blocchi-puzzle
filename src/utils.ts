import { Tetro, Board } from './types';

export const logger = (x: Tetro | Board) => {
  console.log(x.reduce((acc, v) => `${acc} ${v.concat()} \n`, ''));
};
