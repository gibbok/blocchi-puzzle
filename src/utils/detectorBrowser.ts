import { detect } from 'detect-browser';

const getPathRelease = (version: string) => version.split('.')[0];

export const detectorBrowser = (): boolean => {
  const browser = detect();
  if (browser && browser.name && browser.version) {
    const name = browser.name;
    const version = browser.version;
    const path = Number(getPathRelease(version));

    switch (name) {
      case 'chrome':
      case 'firefox':
        return path > 80;
      case 'safari':
        return path > 13;
      case 'opera':
        return path > 62;
      default:
        return false;
    }
  } else {
    return false;
  }
};
