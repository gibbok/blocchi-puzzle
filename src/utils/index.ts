export * from './data.support.test';
export * from './logger';
