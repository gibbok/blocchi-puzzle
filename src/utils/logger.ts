import { Tetro, Board } from '../game/types';

export const logger = (x: Tetro | Board): void => {
  console.log(x.reduce((acc, v) => `${acc} ${v.concat()} \n`, ''));
};
