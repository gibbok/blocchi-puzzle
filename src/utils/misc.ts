import { IO, io } from 'fp-ts/lib/IO';
import { randomInt } from 'fp-ts/lib/Random';

export const getRandomValueFromStringEnum = <T>(strEnum: T): IO<T[keyof T]> => {
  const rndInt = randomInt(0, Object.keys(strEnum).length - 1)();
  const rndValue = Object.keys(strEnum)[rndInt] as keyof T;
  return io.of(strEnum[rndValue]);
};
