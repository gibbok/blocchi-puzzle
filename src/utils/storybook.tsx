import React from 'react';
import GlobalStyle from '../components/GlobalStyle';
import styled from 'styled-components';
import { wood } from '../assets/images';

/**
 * Decorate a story with glocal styles.
 * @param story Story
 */
export const globalStylesDecorator = (story: () => unknown): JSX.Element => (
  <React.Fragment>
    <GlobalStyle />
    {story()}
  </React.Fragment>
);

const Bkg = styled.div`
  background-image: url(${wood});
  width: 100vw;
  height: 100vh;
`;

/**
 * Decorate a story with a page background.
 * @param story Story
 */
export const pageBackground = (story: () => React.ReactNode): JSX.Element => <Bkg>{story()}</Bkg>;
