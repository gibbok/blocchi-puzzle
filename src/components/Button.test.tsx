import React from 'react';
import { Button } from './Button';
import renderer from 'react-test-renderer';
import { noop } from '../utils';

it('should render Button', () => {
  const tree = renderer.create(<Button onClick={noop}>Button</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
