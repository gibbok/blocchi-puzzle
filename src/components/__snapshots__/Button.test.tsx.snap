// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Button 1`] = `
<button
  className="sc-AxjAm bJRPJO"
  data-test="button"
  onClick={[Function]}
>
  <div
    className="sc-AxirZ eJxFBc"
  >
    Button
  </div>
</button>
`;
